build-python.sh

```diff
@@ -352,26 +356,26 @@ if test "${BUILD_WHEEL}" = true; then
     python -m build \
         --wheel \
         --outdir ../dist \
-        ${BUILD_ARGS} \
+        ${BUILD_ARGS:+${BUILD_ARGS}} \
         .
 fi
 
 if test "${INSTALL}" = true; then
     echo "--- installing lightgbm ---"
     cd ../dist
     if test "${BUILD_WHEEL}" = true; then
-        PACKAGE_NAME="lightgbm*.whl"
+        PACKAGE_NAME="$(echo lightgbm*.whl)"
     else
-        PACKAGE_NAME="lightgbm*.tar.gz"
+        PACKAGE_NAME="$(echo lightgbm*.tar.gz)"
     fi
     # ref for use of '--find-links': https://stackoverflow.com/a/52481267/3986677
     pip install \
-        ${PIP_INSTALL_ARGS} \
+        ${PIP_INSTALL_ARGS:+ -o "$PIP_INSTALL_ARGS"} \
```

Comment:
What is the purpose of this change?

Reply:
> I had explained it here-above

I see. I had not seen https://github.com/microsoft/LightGBM/pull/6641#discussion_r1748393161 when I wrote this, maybe it's another one that wasn't published originally because it was part of an unpublished review (as you mentioned in https://github.com/microsoft/LightGBM/pull/6641#issuecomment-2416457468).

I've pushed commits reverting this, for the reasons explained in https://github.com/microsoft/LightGBM/pull/6641#discussion_r1804009897